sentry_sdk/hub.py

```diff
@@ -140,60 +141,31 @@ def main(cls):
 class _ScopeManager:
     def __init__(self, hub):
         # type: (Hub) -> None
-        self._hub = hub
-        self._original_len = len(hub._stack)
-        self._layer = hub._stack[-1]
+        pass
 
     def __enter__(self):
         # type: () -> Scope
-        scope = self._layer[1]
-        assert scope is not None
-        return scope
+        return Scope.get_isolation_scope()
 
     def __exit__(self, exc_type, exc_value, tb):
         # type: (Any, Any, Any) -> None
-        current_len = len(self._hub._stack)
-        if current_len < self._original_len:
-            logger.error(
-                "Scope popped too soon. Popped %s scopes too many.",
-                self._original_len - current_len,
-            )
-            return
-        elif current_len > self._original_len:
-            logger.warning(
-                "Leaked %s scopes: %s",
-                current_len - self._original_len,
-                self._hub._stack[self._original_len :],
-            )
-
-        layer = self._hub._stack[self._original_len - 1]
-        del self._hub._stack[self._original_len - 1 :]
-
-        if layer[1] != self._layer[1]:
-            logger.error(
-                "Wrong scope found. Meant to pop %s, but popped %s.",
-                layer[1],
-                self._layer[1],
-            )
-        elif layer[0] != self._layer[0]:
-            warning = (
-                "init() called inside of pushed scope. This might be entirely "
-                "legitimate but usually occurs when initializing the SDK inside "
-                "a request handler or task/job function. Try to initialize the "
-                "SDK as early as possible instead."
-            )
-            logger.warning(warning)
+        pass
 
 
 class Hub(with_metaclass(HubMeta)):  # type: ignore
-    """The hub wraps the concurrency management of the SDK.  Each thread has
+    """
+    .. deprecated:: 2.0.0
```

Comment:
I think it would make sense to wait to add these deprecation comments until the release where we actually deprecate the hub, and begin emitting deprecation warnings.

Reply:
The idea is to have the deprecation warnings in the apidocs now so people looking at the apidocs know that those apis are deprecated. When we updated our code to not call these apis we add the warning messages. 

@sl0thentr0py do you think this is OK library development practice or should we remove the deprecation from the apidocs too?